sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1059,53 +1059,32 @@ def sdc_pandas_dataframe_drop_impl(df, _func_name, args, columns):
     return sdc_pandas_dataframe_drop_impl(df, _func_name, args, columns)
 
 
-def df_getitem_bool_series_idx_main_codelines(self, idx):
-    """Generate main code lines for df.getitem"""
-    func_lines = ['  self_length = len(get_dataframe_data(self, 0))',
-                  '  trimmed_idx_data = idx._data[:self_length]']
-
-    if isinstance(self.index, types.NoneType):
-        func_lines += ['  self_index = numpy.arange(self_length)']
-    else:
-        func_lines += ['  self_index = self._index']
-
-    results = []
-    for i, col in enumerate(self.columns):
-        res_data = f'res_data_{i}'
-        func_lines += [
-            f'  data_{i} = get_dataframe_data(self, {i})',
-            f'  series = pandas.Series(data_{i}, index=self_index, name="{col}")',
-            f'  {res_data} = series[trimmed_idx_data]',
-        ]
-        results.append((col, res_data))
-
-    data = ', '.join(f'"{col}": {data}' for col, data in results)
-    func_lines += [f'  return pandas.DataFrame({{{data}}}, index=self_index[trimmed_idx_data])']
+def df_length_codelines(self):
+    """Generate code lines to get length of DF"""
+    if self.columns:
+        return ['  length = len(get_dataframe_data(self, 0))']
 
-    return func_lines
+    return ['  length = 0']
 
 
-def df_index_codelines(self):
+def df_index_codelines(self, with_length=False):
```

Comment:
I think it's better for this function to do only index lines generation, so that we just make df_length_codelines call not inside it, but where this func is called with with_lenght=True. This simplifies things, as on the call site you see exactly, that you generate line for lenght, then line for index. 
Alternatively we could use a format string and pass the actual var name that will hold df's len to it as a second argument, i.e.
```
def df_index_codelines(self, length):
...
        func_lines += [f'  res_index = numpy.arange({length})']
```